askama_parser/src/expr.rs

```diff
@@ -15,11 +15,12 @@ use super::{
 
 macro_rules! expr_prec_layer {
     ( $name:ident, $inner:ident, $op:expr ) => {
-        fn $name(i: &'a str) -> IResult<&'a str, Self> {
-            let (i, left) = Self::$inner(i)?;
+        fn $name(i: &'a str, level: Level) -> IResult<&'a str, Self> {
+            let level = level.nest(i)?;
+            let (i, left) = Self::$inner(i, level)?;
             let (i, right) = many0(pair(
                 ws(tag($op)),
-                Self::$inner,
+                |nested| Self::$inner(nested, level),
```

Comment:
I was confused for a moment what `nested` was. I think the identifier `i` would be better. We use it throughout all the code and it is used in all of nom's examples for the input string.

Reply:
Fair enough, renamed them.